### ISSUE TITLE
SSH module takes retry option from task keywords, defaulting to 3

### ISSUE THREAD

**Issue by vladimir-mencl-eresearch** [ansible/ansible#75142]
### Summary

Starting with ansible-core-2.11.0, the `ssh` module operates with a `retry` setting of `3`.

This has undesired impact when using a `sudo` password and making a typo in the password: the module would make 4 connection attempts, attempting a sudo in each connection attempt.

On hosts that report failed sudo attempts via email - this generates 4 emails per Ansible run per host - 4x increase from behaviour with previous version.

I have tracked this to commit 935528e22e5, which changed how the ssh module gets its settings - using `get_option` instead of the ConfigurationManager.

I have worked around this by explicitly setting `retries` on the first task in the playbook - but this is just a workaround.

I hope there would still be a way to set retries for the SSH module (or make it not retry on failed privilege escalation).

Cheers,
Vlad

### Issue Type

Bug Report

### Component Name

ssh

### Ansible Version

```console
ansible [core 2.11.2]
  config file = /Users/vlad/reannz/code/ansible-playbooks/ansible.cfg
  configured module search path = ['/Users/vlad/reannz/code/ansible-playbooks/library']
  ansible python module location = /usr/local/Cellar/ansible/4.1.0/libexec/lib/python3.9/site-packages/ansible
  ansible collection location = /Users/vlad/.ansible/collections:/opt/ansible/collections
  executable location = /usr/local/bin/ansible
  python version = 3.9.5 (default, May  4 2021, 03:29:30) [Clang 11.0.0 (clang-1100.0.33.17)]
  jinja version = 3.0.1
  libyaml = True
```

### Configuration

```console
DEFAULT_BECOME(/Users/vlad/reannz/code/ansible-playbooks/ansible.cfg) = True
DEFAULT_BECOME_ASK_PASS(/Users/vlad/reannz/code/ansible-playbooks/ansible.cfg) = False
DEFAULT_BECOME_METHOD(/Users/vlad/reannz/code/ansible-playbooks/ansible.cfg) = sudo
DEFAULT_BECOME_USER(/Users/vlad/reannz/code/ansible-playbooks/ansible.cfg) = root
DEFAULT_HOST_LIST(/Users/vlad/reannz/code/ansible-playbooks/ansible.cfg) = ['/Users/vlad/reannz/code/ansible-playbooks/environment/production']
```

### OS / Environment

Controller: OSX Mojave 10.14.6
Target: Ubuntu 20.04

### Steps to Reproduce

```ini
# ansible.cfg
[privilege_escalation]
become=True
become_method=sudo
become_user=root
become_ask_pass=False
```
Run:
```
ansible -u $USER -m command -a 'id' 'target-host' -K
```

### Expected Results

I expect sudo attempted only once.

### Actual Results

```console
Sudo is attempted 4 times (as per `/var/log/auth.log` on target host)
```

### Code of Conduct

- [X] I agree to follow the Ansible Code of Conduct

**Comment by ansibot** [comment]
Files identified in the description:
* [`lib/ansible/plugins/connection/ssh.py`](https://github.com/ansible/ansible/blob/devel/lib/ansible/plugins/connection/ssh.py)

If these files are incorrect, please update the `component name` section of the description or use the `!component` bot command.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by samdoran** [comment]
It does look like the default changed from 0 to 3 with the move to config. We should probably change the default back to 0. In the mean time, you can change the retries back to 0 to avoid excessive notifications.

```ini
[ssh_connection]
retries = 0
```

**Comment by vladimir-mencl-eresearch** [comment]
Hi @samdoran ,

Thanks for the quick reply!

I believe what happened is that two options previously independent of each other (`[ssh_connection]` `retries` and task keyword `retries`) got merged into one.

And the `retries` task keyword had the default of `3` already before - just now, that started applying also to the ssh_connection retries.

Cheers,
Vlad

PS: I tried setting the retries in `ansible.cfg` with exactly what you suggest already during my initial troubleshooting, but this is ignored.

My understanding of the code is that as the module calls `get_option`, it always find the value (`3`) in the `_options` dict, because it is already evaluated for the current task (from the task defaults) and set via `set_options`.

So `get_option` never gets to check the value from the module configuration.

**Comment by vladimir-mencl-eresearch** [comment]
PPS: But setting `retries` on the first task in the playbook does work around it....

**Comment by samdoran** [comment]
> I believe what happened is that two options previously independent of each other ([ssh_connection] retries and task keyword retries) got merged into one.

Ok, so it's definitely a bit more complicated than I thought at first. It's a collision of task keywords and config options, as you stated.

**Comment by samdoran** [comment]
@vladimir-mencl-eresearch Can you test with #75155 to see if that fixes the problem for you?

**Comment by vladimir-mencl-eresearch** [comment]
Hi @samdoran ,
Thanks for the quick fix.

Just tested it and works exactly as expected: SSH retries are now controlled by the option and are independent of the task retries.

Tested with env var, Ansible var, module option in both accepted section, with task retries present and absent - does the right think.

Thanks again for the quick fix in #75155 !

Cheers,
Vlad
